### DISCUSSION TITLE
vici

### DISCUSSION THREAD

**Discussion by keskinadem** [strongswan/strongswan#2439] [Q&A]
is there any extended explanation for these commits in this file?(commits https://github.com/strongswan/strongswan/commit/3c5e7eaa88b0068d62a9be139616fb1291e36902, https://github.com/strongswan/strongswan/commit/46ff2688856441dc22daf74d728302ae88b27b81, https://github.com/strongswan/strongswan/commit/ad14f3a60dc7496d061e30920bba3b72dfc6269c)

**Comment by tobiasbrunner** [comment]
What do you mean?

**Reply by keskinadem** [reply]
What is the purpose of these commits? Bug fix or meet some RFC?

**Reply by tobiasbrunner** [reply]
Neither.

**Reply by keskinadem** [reply]
Thanks for answers. I have a certificate and a CA certificate but CA is not issuer of subject. strongswan 6.0.0beta3 tries to verify cert with CA cert in newer version. But it does not try to verify in 5.6.0. That is why I try understand these commits.

**Reply by tobiasbrunner** [reply]
Why would these commits have anything to do with that behavior? And you say the CA certificate is not the issuer, but either it has a matching authorityKeyIdentifier, or if it does not have one at all, a matching subject DN, otherwise, it would not be part of a possible trust chain. strongSwan does not randomly use certificates to verify signatures in others.
